### PR TITLE
layout-products-cards

### PR DESCRIPTION
**Pull request by svyatoslavPy**
- [DEMO LINK](https://svyatoslavPy.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://svyatoslavPy.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/styles/index.scss [hunk 1]
```diff
@@ -1,3 +1,137 @@
...
+    font-weight: 500;
+    line-height: 18px;
+    width: 166px;
+  }
+
+  &__description {
+    margin: 0;
+    color: $card-description-text;
+    font-size: 10px;
+    font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:40]: default
```suggestion
```

File: src/styles/index.scss [hunk 2]
```diff
@@ -1,3 +1,137 @@
...
+    line-height: 14px;
+    margin-top: 4px;
+  }
+
+  &__reviews {
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:46]: reset for tags you need

File: src/styles/index.scss [hunk 3]
```diff
@@ -1,3 +1,137 @@
...
+    font-size: 10px;
+    font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:48]:
```suggestion
```

File: src/styles/index.scss [hunk 4]
```diff
@@ -1,3 +1,137 @@
...
+  }
+
+  &__prices {
+    @include flex-space-between;
+
+    margin-top: 24px;
+  }
+
+  &__text {
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:61]:
```suggestion
```

File: src/styles/index.scss [hunk 5]
```diff
@@ -1,3 +1,137 @@
...
+    color: $card-primary-text;
+    font-size: 12px;
+    font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:64]:
```suggestion
```

File: src/styles/index.scss [hunk 6]
```diff
@@ -1,3 +1,137 @@
...
+    line-height: 18px;
+  }
+
+  &__price {
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:69]:
```suggestion
```

File: src/styles/index.scss [hunk 7]
```diff
@@ -1,3 +1,137 @@
...
+    font-size: 16px;
+    font-weight: 700;
+    line-height: 18px;
+    color: $card-price-text;
+  }
+
+  &__button {
+    box-sizing: border-box;
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:78]:
```suggestion
```

File: src/styles/index.scss [hunk 8]
```diff
@@ -1,3 +1,137 @@
...
+
+    @include flex-center;
+
+    border: none;
+    width: 166px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:83]: full width
```suggestion
    width: 100%;
```

File: src/styles/index.scss [hunk 9]
```diff
@@ -1,3 +1,137 @@
...
+    height: 40px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:84]: it's not a good practice to hardcode height for elements with content inside. use padding instead

File: src/styles/index.scss [hunk 10]
```diff
@@ -1,3 +1,137 @@
...
+    font-size: 14px;
+    font-weight: 700;
+    line-height: 16px;
+    text-transform: uppercase;
+    margin-top: 16px;
+    cursor: pointer;
+
+    &:hover {
+      background: #FFF;
+      color: #00ACDC;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:97]: move repeated colors to variables

File: src/styles/index.scss [hunk 11]
```diff
@@ -1,3 +1,137 @@
...
+    background-position: center;
+    margin-left: 4px;
+
+    &:first-child {
+      margin-left: 0;
+    }
+  }
+
+  /* Stars :active */
+  &--4 {
```
Comment (Viktor-Kost) [inline src/styles/index.scss:129]: add styles for all possible modifiers

File: src/styles/index.scss [hunk 12]
```diff
@@ -1,3 +1,137 @@
...
+  &__inner {
+    @include flex-space-between;
+
+    margin-top: 16px;
+  }
+
+}
+
+
+.stars {
```
Comment (Viktor-Kost) [inline src/styles/index.scss:112]: move stars styles to an own file

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 13]
```diff
@@ -6,8 +6,43 @@
...
+        src="images/imac.jpeg"
+        alt="product"
+      >
+      <h2 class="card__title">
+        APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
+      </h2>
+      <p class="card__description">
+        Product code: 195434
+      </p>
+      <div class="card__inner">
```
Comment (volodymyr-soltys97) [inline src/index.html:26]: Add the blank line between the blocks
```suggestion
      >
      
      <h2 class="card__title">
        APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
      </h2>
      
      <p class="card__description">
        Product code: 195434
      </p>
      
      <div class="card__inner">
```

File: src/styles/blocks/card.scss [hunk 14]
```diff
@@ -0,0 +1,99 @@
+.card {
+  width: 200px;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: $card-bg;
+  padding: 32px 16px 16px;
+
+
+
+  &__picture {
```
Comment (volodymyr-soltys97) [inline src/styles/blocks/card.scss:10]:
```suggestion
  padding: 32px 16px 16px;

  &__picture {
```

File: src/styles/blocks/card.scss [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+    font-weight: 500;
+    line-height: 18px;
+    width: 166px;
+  }
+
+  &__description {
+    margin: 0;
+    color: $card-description-text;
+    font-size: 10px;
+    font-weight: 400;
```
Comment (volodymyr-soltys97) [inline src/styles/blocks/card.scss:30]: Default value, so you can just remove it everywhere
```suggestion
```

File: src/styles/utils/variables.scss [hunk 16]
```diff
@@ -0,0 +1,11 @@
+$card-bg: #FFF;
+$card-reviews-text: #060B35;
+$card-price-text: #060B35;
+$card-title-text: #060B35;
```
Comment (volodymyr-soltys97) [inline src/styles/utils/variables.scss:4]: Why are you creating multiple variables with the same value? If only one variable can be used everywhere
```suggestion
$card-reviews-text: #060B35;
```

File: src/styles/utils/variables.scss [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+$card-description-text: #616070;
+$card-primary-text: #616070;
```
Comment (volodymyr-soltys97) [inline src/styles/utils/variables.scss:6]: The same
```suggestion
$card-description-text: #616070;
```

File: src/styles/utils/variables.scss [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+$card-button-bg: #00ACDC;
+$card-button-color: #00ACDC;
```
Comment (volodymyr-soltys97) [inline src/styles/utils/variables.scss:8]: Same
```suggestion
$card-button-bg: #00ACDC;
```

File: src/styles/utils/variables.scss [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+$card-button-bg-hover: #fff;
+$card-button-text: #fff;
```
Comment (volodymyr-soltys97) [inline src/styles/utils/variables.scss:10]: And here
```suggestion
$card-button-bg-hover: #fff;
```

File: src/styles/utils/variables.scss [hunk 20]
```diff
@@ -0,0 +1,11 @@
+$card-bg: #FFF;
```
Comment (volodymyr-soltys97) [inline src/styles/utils/variables.scss:1]:
```suggestion
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Hi, your styles doesn't work, you need to fix it
image

**Review by TarasHoliuk — APPROVED** [review]
Generally, the code looks good, so approved :heavy_check_mark: 

But The demo link looks not good at all:

image

Next time _make sure_  the deployed solution works. Otherwise ask help in the chat to fix it

**Pull request closed without merging**